### PR TITLE
Add integration tests for Nodegroups and Addons

### PR DESCRIPTION
**Pull request by RedbackThomson**
Description of changes:
Add create/delete tests for Nodegroup and Addon types

By submitting this pull request, I confirm that my contribution is made under
the terms of the Apache 2.0 license.

### CODE REVIEW

File: go.local.mod [hunk 1]
```diff
@@ -5,7 +5,7 @@ go 1.14
 replace github.com/aws-controllers-k8s/runtime => ../runtime
 
 require (
-	github.com/aws-controllers-k8s/runtime v0.12.0
+	github.com/aws-controllers-k8s/runtime v0.15.2
```
Comment (jaypipes) [inline go.local.mod:8]: I believe you should add this change in a separate commit/PR.

**Review by jaypipes — CHANGES_REQUESTED** [review]
Let's have the runtime version update changes in a separate PR please, thank you @RedbackThomson!

**Comment by RedbackThomson** [comment]
> Let's have the runtime version update changes in a separate PR please, thank you @RedbackThomson!

It's just for the local Go mod file. It had a version mismatch compared to the "real" Go mod.

**Comment by jaypipes** [comment]
> > Let's have the runtime version update changes in a separate PR please, thank you @RedbackThomson!
> 
> It's just for the local Go mod file. It had a version mismatch compared to the "real" Go mod.

Apologies, I missed that it was the local mod files...

**Review by jaypipes — APPROVED** [review]
:+1:

**Comment by jaypipes** [comment]
@RedbackThomson are the test failures just a timeout issue?

**Comment by RedbackThomson** [comment]
> @RedbackThomson are the test failures just a timeout issue?

Not exactly sure what I'm seeing here. I had multiple successful test runs in my local account. 

From the tests for `Nodegroup`:
```
Waiter encountered a terminal failure state: For expression "nodegroup.status" we matched expected path: "CREATE_FAILED"
```
Seems to imply that our node groups are failing to create?

From the tests for `Cluster`:
```
Cluster 'simple-cluster-mk8ub3sbxwegxqzj4' was not deleted from EKS
```
This test has been running for weeks without seeing problems, so don't see why we would have an issue here either.

**Pull request merged**